### PR TITLE
bumping master to Swift 5.0 

### PR DESCRIPTION
**Pull request by najacque**
This PR bumps the Swift compiler version to 5.0 for the swift master branch.

Resolves rdar://problem/43523881

### CODE REVIEW

**Comment by najacque** [comment]
@swift-ci please test

**Comment by najacque** [comment]
Looks like I need to also update getEffectiveLanguageVersion, and per @jrose-apple should not change the  crash recovery test...

**Comment by jrose-apple** [comment]
Ah, here we go: #13843 is what I was looking at back before we confirmed that 4.2 was going to happen. It doesn't answer all the questions about what we should do this time, but it shows what I had to touch then.

**Comment by najacque** [comment]
@swift-ci please smoke test

File: lib/Basic/Version.cpp [hunk 1]
```diff
@@ -317,28 +317,25 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
...
 #else
     return Version{3, 4};
 #endif
   case 4:
-    static_assert(SWIFT_VERSION_MAJOR == 4,
-                  "getCurrentLanguageVersion is no longer correct here");
     // Version '4' on its own implies '4.1.50'.
     if (size() == 1)
       return Version{4, 1, 50};
     return Version::getCurrentLanguageVersion();
```
Comment (jrose-apple) [inline lib/Basic/Version.cpp]: This line here's no longer going to be right. Maybe the whole section should be "If Version >= 4.2, return 4.2, else return 4.1.50".

**Comment by tkremenek** [comment]
@najacque I'd be happy to take this over at this point, since I did the surgery here for the 4.2 version bump.  Just let me know.

**Comment by najacque** [comment]
@tkremenek, sounds good,  I got side tracked with other stuff, so please go ahead.

**Comment by tkremenek** [comment]
Subsumed by https://github.com/apple/swift/pull/20374

**Pull request closed without merging**